src/ui-components/input-field/input-field.spec.js

```diff
@@ -0,0 +1,315 @@
+          value: undefined,
+        },
+        expectedResults: {
+          id: "testId123",
+          type: "text",
+          required: undefined,
+          value: "",
+        }
+      },
+      {
+        description: "when required is true",
+        givenData: {
+          id: "",
+          type: undefined,
+          required: true,
+          value: undefined,
+        },
+        expectedResults: {
+          id: "",
+          type: "text",
+          required: true,
+          value: "",
+        }
+      },
+      {
+        description: "when required is false",
+        givenData: {
+          id: "",
+          type: undefined,
+          required: false,
+          value: undefined,
+        },
+        expectedResults: {
+          id: "",
+          type: "text",
+          required: false,
+          value: "",
+        }
+      }
+    ]
+
+    testCases.forEach(testCase => {
+      it(testCase.description, () => {
+        testInputFieldProps(testCase.givenData, testCase.expectedResults);
+      });
+    });
+
+  });
+
+    describe("validation messages", () => {
+      it("are rendered when provided only one", () => {
+        const testValidationMessages = ["validation message"]
+        const wrapper = shallow(<InputField validationMessages={testValidationMessages} />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeTruthy();
+        expect(wrapper.find(`.${CSS_CLASSES.VALIDATION_MESSAGE}`).text().trim()).toEqual(testValidationMessages[0]);
+      });
+      it("are rendered when provided when provided multiple", () => {
+        const testValidationMessages = ["validation message", "validation message 2", "validation message 3"];
+        const wrapper = shallow(<InputField validationMessages={testValidationMessages} />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeTruthy();
+        expect(wrapper.find(`.${CSS_CLASSES.VALIDATION_MESSAGE}`).length).toEqual(3);
+      });
+      it("are not rendered when not provided", () => {
+        const wrapper = shallow(<InputField />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeFalsy();
+      });
+    });
+
+    describe("label", () => {
+      it("is rendered when provided", () => {
+        const testLabel = "this is label";
+        const wrapper = shallow(<InputField label={testLabel} />);
+
+        expect(wrapper.exists("label")).toBeTruthy();
+        expect(wrapper.find("label").text().trim()).toEqual(testLabel);
+      });
+      it("is not rendered when not provided", () => {
+        const wrapper = shallow(<InputField />);
+
+        expect(wrapper.exists("label")).toBeFalsy();
+      });
+      it("has label lowered class applied when field is empty", () => {
+        const testLabel = "this is label";
+        const wrapper = shallow(<InputField label={testLabel} />);
+
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_LOWERED)).toEqual(true);

```

Comment:
These multiple expects could be refactored into a single one using the enzyme prop() api like:

```
const labelClass = wrapper.find("label").prop("className");
const expectedClass = CSS_CLASSES.LABEL_LOWERED;
expect(labelClass).toBe(expectedClass);
```

- [ ] Refactor multiple expects into single one.